tests/main/interfaces-browser-support/task.yaml

```diff
@@ -0,0 +1,138 @@
+        chown test:12345 "$owned_file"
+        su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $owned_file" test | MATCH test
+        su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd touch $owned_file" test
+    done
+
+    echo "And the snap is able to access readable files"
+    for readable_file in $READABLE_FILES; do
+        parent_dir=$(dirname $readable_file)
+        if [ ! -d $parent_dir ]; then
+            if mkdir -p $parent_dir; then
+                echo "$parent_dir" >> created_dirs
+            else
+                echo "$parent_dir couldn't be created, write-only partition?"
+                continue
+            fi
+        fi
+        echo "test" > "$readable_file"
+        su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $readable_file" test | MATCH test
+    done
+
+    if [ "$ALLOW_SANDBOX" = "true" ]; then
+        for readable_file in $READABLE_WITH_SANDBOX_FILES; do
+            parent_dir=$(dirname $readable_file)
+            if [ ! -d $parent_dir ]; then
+                mkdir -p $parent_dir
+                echo "$parent_dir" >> created_dirs
+            fi
+            echo "test" > "$readable_file"
+            su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $readable_file" test | MATCH test
+        done
+    fi
+
+    if [ "$(snap debug confinement)" = strict ] ; then
+        echo "When the plug is disconnected"
+        snap disconnect browser-support-consumer:browser-support
+        snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+        echo "Then the snap is not able to access tmp"
+        if su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd ls /var/tmp/ 2>${PWD}/tmpdir-access.err" test; then
+            echo "Expected error with disconnected plug didn't happen"
+            exit 1
+        fi
+        MATCH "Permission denied" < tmpdir-access.err
+        if su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat /var/tmp/etilqs_test 2>${PWD}/tmpfile-read.err" test; then
+            echo "Expected error with disconnected plug didn't happen"
+            exit 1
+        fi
+        MATCH "Permission denied" < tmpfile-read.err
+
+        for owned_file in $OWNED_FILES; do
+            if su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $owned_file 2>${PWD}/owned-read.err" test; then
+                echo "Expected error with disconnected plug didn't happen"
+                exit 1
+            fi
+            MATCH "Permission denied" < owned-read.err
+        done
+        for readable_file in $READABLE_FILES; do
+            if [ -f "$readable_file" ] && su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $readable_file 2>${PWD}/readable-read.err" test; then
+                echo "Expected error with disconnected plug didn't happen"
+                exit 1
+            fi
+            MATCH "Permission denied" < readable-read.err
+        done
+        if [ "$ALLOW_SANDBOX" = "true" ]; then
```

Comment:
It would be nice to have something checking the false case, otherwise we're really testing one case only. Imagine that the backend was implemented ignoring that flag altogether, and simply implementing the one behavior matching the true case &ndash; these tests would still pass.